src/lib/jsonParseV2/parseSubscription.cpp

```diff
@@ -355,101 +299,75 @@ static std::string parseSubject(ConnectionInfo* ciP, SubscribeContextRequest* sc
 * parseEntitiesVector -
 *
 */
-static std::string parseEntitiesVector(ConnectionInfo* ciP, EntityIdVector* eivP, const Value& entities)
+static std::string parseEntitiesVector(ConnectionInfo* ciP, std::vector<EntID>* eivP, const Value& entities)
 {
   if (!entities.IsArray())
   {
-    alarmMgr.badInput(clientIp, "subject entities is not an array");
-    OrionError oe(SccBadRequest, "subject entities is not an array");
-
-    return oe.render(ciP, "");
+    return error(ciP, "subject entities is not an array");
   }
 
   for (Value::ConstValueIterator iter = entities.Begin(); iter != entities.End(); ++iter)
   {
     if (!iter->IsObject())
     {
-      alarmMgr.badInput(clientIp, "subject entities element is not an object");
-      OrionError oe(SccBadRequest, "subject entities element is not an object");
-
-      return oe.render(ciP, "");
+     return error(ciP, "subject entities element is not an object");
     }
     if (!iter->HasMember("id") && !iter->HasMember("idPattern") && !iter->HasMember("type"))
     {
-      alarmMgr.badInput(clientIp, "subject entities element has not id/idPattern nor type");
-      OrionError oe(SccBadRequest, "subject entities element has not id/idPattern");
-
-      return oe.render(ciP, "");
+      return error(ciP, "subject entities element has not id/idPattern nor type");
     }
 
     if (iter->HasMember("id") && iter->HasMember("idPattern"))
     {
-      alarmMgr.badInput(clientIp, "subject entities element has id and idPattern");
-      OrionError oe(SccBadRequest, "subject entities element has id and idPattern");
-
-      return oe.render(ciP, "");
+      return error(ciP, "subject entities element has id and idPattern");
     }
 
-
     std::string  id;
+    std::string  idPattern;
     std::string  type;
-    std::string  isPattern  = "false";
 
-    if (iter->HasMember("id"))
     {
-      if ((*iter)["id"].IsString())
+      Opt<std::string> idOpt = getStringOpt(*iter,"id", "subject entities element id");
+      if (!idOpt.ok())
       {
-        id = (*iter)["id"].GetString();
+        return error(ciP, idOpt.error);
       }
-      else
-      {
-        alarmMgr.badInput(clientIp, "subject entities element id is not a string");
-        OrionError oe(SccBadRequest, "subject entities element id is not a string");
+      id = idOpt.value;
 
-        return oe.render(ciP, "");
+      if (id.empty())
+      {
+        idPattern = ".*";
       }
```

Comment:
So, if `id` is given as empty string, we change it to **all** entities?
I didn't know, but it makes sense, analogous to empty `type`.
Is this something new?